packages/features/availability/lib/getUserAvailability.ts

```diff
@@ -370,6 +370,116 @@ export class UserAvailabilityService {
     // if no schedules set by or for a user, use fallbackSchedule
     const schedule = potentialSchedule ?? fallbackSchedule;
 
+    if (
+      !(
+        schedule?.availability ||
+        (eventType?.availability.length ? eventType.availability : user.availability)
+      )
+    ) {
+      throw new HttpError({ statusCode: 400, message: ErrorCode.AvailabilityNotFoundInSchedule });
+    }
+
+    const availability = (
+      schedule?.availability || (eventType?.availability.length ? eventType.availability : user.availability)
+    ).map((a) => ({
+      ...a,
+      userId: user.id,
+    }));
+
+    const isTimezoneSet = Boolean(potentialSchedule && potentialSchedule.timeZone !== null);
+
+    // this timezone is synced with google/outlook calendars timezone usingg delegated credentials
+    // it's a fallback for delegated credentials users who want to sync their timezone with third party calendars
+    const calendarTimezone = !isTimezoneSet ? await this.getTimezoneFromDelegatedCalendars(user) : null;
```

Comment:
nit: Maybe we take this opportunity to move the newly moved code to a separate handler/fn, makes __getUserAvailability_ fn more manageable